src/app/targetDownloader.js

```diff
@@ -95,6 +96,13 @@ createNameSpace("realityEditor.app.targetDownloader");
         const navmesh = evt.data.navmesh;
         const objectID = evt.data.objectID;
         navmeshResolution = evt.data.heatmapResolution;
+        for (let i = 0; i < window.localStorage.length; i++) {
```

Comment:
This is okay for now, but it would be better to only start removing items if setItem fails, i.e.:
```javascript
try {
       window.localStorage.setItem(`realityEditor.navmesh.${objectID}`, JSON.stringify(navmesh));
} catch (e) {
       console.error('Out of memory, clearing navmesh cache', e);
       for (let i = 0; i < window.localStorage.length; i++) {
            const key = window.localStorage.key(i);
            if (key.includes("realityEditor.navmesh.") && !key.includes(`${objectID}`)) {
                window.localStorage.removeItem(key);
                i--;
            }
       }
      window.localStorage.setItem(`realityEditor.navmesh.${objectID}`, JSON.stringify(navmesh));
}
```

Reply:
Makes sense, I'll include it in the next PR. Thx James!